bpf/lib/nat.h

```diff
@@ -558,22 +560,20 @@ static __always_inline void snat_v4_init_tuple(const struct iphdr *ip4,
 }
 
 /* The function contains a core logic for deciding whether an egressing packet
- * has to be SNAT-ed. Currently, the function targets the following flows:
- *
- *	- From pod to outside to masquerade requests
- *	  when --enable-bpf-masquerade=true.
- *	- From host to outside to track (and masquerade) flows which
- *	  can conflict with NodePort BPF.
+ * has to be SNAT-ed, filling the relevant state in the target parameter if
+ * that's the case.
  *
- * The function sets "addr" to the SNAT IP addr, and "from_endpoint" to true
- * if the packet is sent from a local endpoint.
+ * The function will set:
+ * - target->addr to the SNAT IP address
+ * - target->from_local_endpoint to true if the packet is sent from a local endpoint
+ * - target->egress_gateway to true if the packet should be SNAT-ed because of
+ *   an egress gateway policy
  *
- * Callers should treat contents of "from_endpoint" and "addr" as undetermined,
- * if function returns false.
+ * The function will return true if the packet should be SNAT-ed, false
+ * otherwise.
  */
-static __always_inline bool snat_v4_needed(struct __ctx_buff *ctx,
-					   struct ipv4_nat_target *target,
-					   bool *from_endpoint __maybe_unused)
+static __always_inline bool snat_v4_prepare_state(struct __ctx_buff *ctx,
```

Comment:
It pains me to suggest this because my IPv6 masquerading PR rebase will hurt, but: should we also rename the IPv6 function accordingly?

Reply:
I looked at renaming this function as well, but it's not really preparing any state, it's just running a couple of checks on DSR and source address :thinking: so fine for me to rename it for consistency, but the name may be a bit misleading (at least for now)

Reply:
Agree. I don't feel strongly either way.

Reply:
Let's leave it as it is then, the `/* See snat_v4_prepare_state(). */` comment should be already enough for now to cross reference it the v4 implementation